src/main/java/org/zalando/nakadi/service/CursorOperationsService.java

```diff
@@ -195,8 +196,13 @@ private NakadiCursor moveBack(final ShiftedNakadiCursor cursor) throws InvalidCu
     }
 
     private long numberOfEventsBeforeCursor(final NakadiCursor initialCursor) {
-        final TopicRepository topicRepository = getTopicRepository(initialCursor.getTimeline());
-        return topicRepository.numberOfEventsBeforeCursor(initialCursor);
+        final Storage.Type storageType = initialCursor.getTimeline().getStorage().getType();
+        switch (storageType) {
+            case KAFKA:
+                return KafkaCursor.toKafkaOffset(initialCursor.getOffset());
```

Comment:
I understand the fix you did but I don't exactly understand how the code works at a bigger scale. 
E.g. why we need to know number of events before cursor that is located in timeline which is already deleted? Or do we need to be able to calculate the distance even between expired offsets/cursors?

Reply:
Well, we could of course improve this, but in my opinion this falls into the scope of the to-be-created ticket about refactoring this portion of the code. Yesterday, some of us discussed that the cursors code should probably be refactored entirely, so this PR is just a quick fix.

Since it is a quick fix, I prefer to keep it as small as possible.